docs/source/deployment-experience.md

```diff
@@ -0,0 +1,135 @@
+# Deployment Experience
+
+Several facilities deploy Bluesky across many instruments (up to 25 or so).
+In this document we summarize lessons learned from that experience. At the
+moment, only NSLS-II (Bluesky's facility of origin) is represented, but we
+hope to add comments from other facilities in the future.
+
+## NSLS-II
+
+### Python environment management
+
+We deploy conda environments, verioned by operating cycle, in a central (NFS)
+software mount. The instrument staff and users have only read-only access to
+these environments. The environments contains the union of all requirements
+across all instruments: they include core scientific Python libraries (e.g.
+scipy), Bluesky libraries, and other more instrument-specific libraries, both
+internally and externally developed.
+
+The environments read-only in order to avoid accidentally breaking changes.
+To support short-term patching and minor variations betweeen instruments,
+we add an instrument-specific directory to the `PYTHONPATH` at each
+instrument respectively, granting instrument staff read/write permission
+on that directory. This enables them to install extra packages and upgrade
+or downgrade existing ones as needed, without the risk of unintended
+consequences for other instruments.
+
+The conda environments are build via a
+[CI service](https://github.com/nsls2-conda-envs),
+hosted on [Zenodo](https://zenodo.org/record/6555136), and
+deployed via conda-pack. Using conda-pack speeds up installation and
+improves reliability.
+
+We recommend:
+
+* Create centrally-managed read only conda enviornments.
+* Use `PYTHONPATH` to "overlay" custom additions or version changes.
```

Comment:
It's comforting to see how many deployment similarities NSLS-II has with LCLS. This section here that I'm attaching this comment to is exactly what we do at LCLS- centrally managed conda + PYTHONPATH for custom edits as needed for an experiment when it can't wait for a formal release cycle.

Reply:
Our NFS has been historically sometimes slow, so in addition we've been using conda-pack to deploy copies of the central environments locally to SSDs on operator consoles.